arboretum/permissions/checks/test_org_permissions.py

```diff
@@ -0,0 +1,109 @@
+# -*- mode:python; coding:utf-8 -*-
+# Copyright (c) 2021 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Repository organization permissions check."""
+
+from compliance.check import ComplianceCheck
+from compliance.evidence import DAY, ReportEvidence, evidences
+from compliance.utils.data_parse import get_sha256_hash
+
+
+class OrgPermissionsCheck(ComplianceCheck):
+    """Monitor repository organization permissions."""
+
+    @property
+    def title(self):
+        """
+        Return the title of the checks.
+
+        :returns: the title of the checks
+        """
+        return 'Report on Repository Organization Permissions'
+
+    @classmethod
+    def setUpClass(cls):
+        """Initialize the check object with configuration settings."""
+        cls.config.add_evidences(
+            [
+                ReportEvidence(
+                    'org_permissions.md',
+                    'permissions',
+                    DAY,
+                    'Repository Permissions.'
+                )
+            ]
+        )
+        return cls
+
+    def test_org_permissions(self):
+        """Check the access to organization repositories."""
+        orgs = self.config.get('org.permissions.org_integrity.orgs')
+        for org in orgs:
+            host = org['url'].rsplit('/', 1)
+            service = 'gh'
+            if 'gitlab' in host:
+                service = 'gl'
+            elif 'bitbucket' in host:
+                service = 'bb'
+            url_hash = get_sha256_hash([org['url']], 10)
+            path = 'raw/permissions'
+            evidence_paths = {
+                'direct': f'{path}/{service}_direct_collaborators_'
+                + f'{url_hash}.json',
+                'outside': f'{path}/{service}_outside_collaborators_'
+                + f'{url_hash}.json',
+                'forks': f'{path}/{service}_forks_{url_hash}.json'
+            }
+            with evidences(self, evidence_paths) as ev:
```

Comment:
If you handled the split correctly to get the `host` and `org_name` as is done in [the check](https://github.com/ComplianceAsCode/auditree-arboretum/blob/main/arboretum/permissions/checks/test_org_collaborators.py#L57-L70) that you should be using as a guide then you could also have the `with evidences(...)` context manager at the same level as the `for org in orgs:` loop.  Unless there's a good reason to have the context manager imbedded in the `for` loop?

Reply:
I need the nested loop because I have to handle multiple evidence files non only one as in the other check. The evidence path key must be the evidence type(direct, outside or forks)